src/include/function/string/functions/substr_function.h

```diff
@@ -17,7 +17,7 @@ struct SubStr {
         bool isAscii = true;
         int64_t startPos = start - 1;
         int64_t endPos = std::min(srcStr.size(), (size_t)(startPos + len));
-        if (startPos >= endPos) {
+        if (startPos >= endPos || src.isOutOfBounds(startPos) || src.isOutOfBounds(endPos - 1)) {
```

Comment:
I would not add a new function isOutOfBound
I don't think endpos can be out of bound since the value is calculated by: std::min(srcStr.size(), (size_t)(startPos + len));

Reply:
That's true. I'll just check startPos without a function.